File: Nx3CS/bin/test/src/nx3/NVoice.cs

#pragma warning disable 109, 114, 219, 429, 168, 162
namespace nx3{
	public  class NVoice : global::haxe.lang.HxObject {
		public    NVoice(global::haxe.lang.EmptyObject empty){
			unchecked {
				#line 11 "F:\\nx3\\src\\nx3\\NVoice.hx"
				{
				}
				
			}
			#line default
		}
		
		
		public    NVoice(global::Array<object> notes, global::nx3.EVoiceType type, global::nx3.EDirectionUAD direction){
			unchecked {
				#line 20 "F:\\nx3\\src\\nx3\\NVoice.hx"
				global::nx3.NVoice.__hx_ctor_nx3_NVoice(this, notes, type, direction);
			}
			#line default
		}
		
		
		public static   void __hx_ctor_nx3_NVoice(global::nx3.NVoice __temp_me73, global::Array<object> notes, global::nx3.EVoiceType type, global::nx3.EDirectionUAD direction){
			unchecked {
				#line 23 "F:\\nx3\\src\\nx3\\NVoice.hx"
				if (( ( notes == default(global::Array<object>) ) || ( notes.length == 0 ) )) {
					#line 25 "F:\\nx3\\src\\nx3\\NVoice.hx"
					__temp_me73.nnotes = new global::Array<object>(new object[]{});
					__temp_me73.type = global::nx3.EVoiceType.Barpause(0);
				}
				 else {
					#line 30 "F:\\nx3\\src\\nx3\\NVoice.hx"
					__temp_me73.nnotes = notes;
					if (( type != default(global::nx3.EVoiceType) )) {
						#line 31 "F:\\nx3\\src\\nx3\\NVoice.hx"
						__temp_me73.type = type;
					}
					 else {
						#line 31 "F:\\nx3\\src\\nx3\\NVoice.hx"
						__temp_me73.type = global::nx3.EVoiceType.Normal;
					}
					
				}
				
				#line 34 "F:\\nx3\\src\\nx3\\NVoice.hx"
				{
					#line 34 "F:\\nx3\\src\\nx3\\NVoice.hx"
					int _g = 0;
					#line 34 "F:\\nx3\\src\\nx3\\NVoice.hx"
					while (( _g < notes.length )){
						#line 34 "F:\\nx3\\src\\nx3\\NVoice.hx"
						global::nx3.NNote note = ((global::nx3.NNote) (notes[_g]) );
						#line 34 "F:\\nx3\\src\\nx3\\NVoice.hx"
						 ++ _g;
						#line 34 "F:\\nx3\\src\\nx3\\NVoice.hx"
						note.nvoice = __temp_me73;
					}
					
				}
				
				#line 36 "F:\\nx3\\src\\nx3\\NVoice.hx"
				if (( direction != default(global::nx3.EDirectionUAD) )) {
					#line 36 "F:\\nx3\\src\\nx3\\NVoice.hx"
					__temp_me73.direction = direction;
				}
				 else {
					#line 36 "F:\\nx3\\src\\nx3\\NVoice.hx"
					__temp_me73.direction = global::nx3.EDirectionUAD.Auto;
				}
				
			}
			#line default
		}
		
		
		public static  new object __hx_createEmpty(){
			unchecked {
				#line 11 "F:\\nx3\\src\\nx3\\NVoice.hx"
				return new global::nx3.NVoice(((global::haxe.lang.EmptyObject) (global::haxe.lang.EmptyObject.EMPTY) ));
			}
			#line default
		}
		
		
		public static  new object __hx_create(global::Array arr){
			unchecked {
				#line 11 "F:\\nx3\\src\\nx3\\NVoice.hx"
				return new global::nx3.NVoice(((global::Array<object>) (global::Array<object>.__hx_cast<object>(((global::Array) (arr[0]) ))) ), ((global::nx3.EVoiceType) (arr[1]) ), ((global::nx3.EDirectionUAD) (arr[2]) ));
			}
			#line default
		}
		
		
		public  global::nx3.EDirectionUAD direction;
		
		public  global::Array<object> nnotes;
		
		public  global::nx3.EVoiceType type;
		
		public  global::nx3.NPart npart;
		
		public virtual   object iterator(){
			unchecked {
				#line 39 "F:\\nx3\\src\\nx3\\NVoice.hx"
				return ((object) (new global::_Array.ArrayIterator<object>(((global::Array<object>) (this.nnotes) ))) );
			}
			#line default
		}
		
		
		public  int length;
		
		public virtual   int get_length(){
			unchecked {
				#line 41 "F:\\nx3\\src\\nx3\\NVoice.hx"
				return this.nnotes.length;
			}
			#line default
		}
		
		
		public virtual   global::nx3.NNote getNNote(int idx){
			unchecked {
				#line 44 "F:\\nx3\\src\\nx3\\NVoice.hx"
				if (( ( idx < 0 ) || ( idx > this.nnotes.length ) )) {
					#line 44 "F:\\nx3\\src\\nx3\\NVoice.hx"
					return default(global::nx3.NNote);
				}
				 else {
					#line 44 "F:\\nx3\\src\\nx3\\NVoice.hx"
					return ((global::nx3.NNote) (this.nnotes[idx]) );
				}
				
			}
			#line default
		}
		
		
		public override   double __hx_setField_f(string field, int hash, double @value, bool handleProperties){
			unchecked {
				#line 11 "F:\\nx3\\src\\nx3\\NVoice.hx"
				switch (hash){
					case 520590566:
					{
						#line 11 "F:\\nx3\\src\\nx3\\NVoice.hx"
						this.length = ((int) (@value) );
						#line 11 "F:\\nx3\\src\\nx3\\NVoice.hx"
						return @value;
					}
					
					
					default:
					{
						#line 11 "F:\\nx3\\src\\nx3\\NVoice.hx"
						return base.__hx_setField_f(field, hash, @value, handleProperties);
					}
					
				}
				
			}
			#line default
		}
		
		
		public override   object __hx_setField(string field, int hash, object @value, bool handleProperties){
			unchecked {
				#line 11 "F:\\nx3\\src\\nx3\\NVoice.hx"
				switch (hash){
					case 520590566:
					{
						#line 11 "F:\\nx3\\src\\nx3\\NVoice.hx"
						this.length = ((int) (global::haxe.lang.Runtime.toInt(@value)) );
						#line 11 "F:\\nx3\\src\\nx3\\NVoice.hx"
						return @value;
					}
					
					
					case 543535969:
					{
						#line 11 "F:\\nx3\\src\\nx3\\NVoice.hx"
						this.npart = ((global::nx3.NPart) (@value) );
						#line 11 "F:\\nx3\\src\\nx3\\NVoice.hx"
						return @value;
					}
					
					
					case 1292432058:
					{
						#line 11 "F:\\nx3\\src\\nx3\\NVoice.hx"
						this.type = ((global::nx3.EVoiceType) (@value) );
						#line 11 "F:\\nx3\\src\\nx3\\NVoice.hx"
						return @value;
					}
					
					
					case 453807379:
					{
						#line 11 "F:\\nx3\\src\\nx3\\NVoice.hx"
						this.nnotes = ((global::Array<object>) (global::Array<object>.__hx_cast<object>(((global::Array) (@value) ))) );
						#line 11 "F:\\nx3\\src\\nx3\\NVoice.hx"
						return @value;
					}
					
					
					case 272654911:
					{
						#line 11 "F:\\nx3\\src\\nx3\\NVoice.hx"
						this.direction = ((global::nx3.EDirectionUAD) (@value) );
						#line 11 "F:\\nx3\\src\\nx3\\NVoice.hx"
						return @value;
					}
					
					
					default:
					{
						#line 11 "F:\\nx3\\src\\nx3\\NVoice.hx"
						return base.__hx_setField(field, hash, @value, handleProperties);
					}
					
				}
				
			}
			#line default
		}
		
		
		public override   object __hx_getField(string field, int hash, bool throwErrors, bool isCheck, bool handleProperties){
			unchecked {
				#line 11 "F:\\nx3\\src\\nx3\\NVoice.hx"
				switch (hash){
					case 1775117066:
					{
						#line 11 "F:\\nx3\\src\\nx3\\NVoice.hx"
						return ((global::haxe.lang.Function) (new global::haxe.lang.Closure(((object) (this) ), ((string) ("getNNote") ), ((int) (1775117066) ))) );
					}
					
					
					case 261031087:
					{
						#line 11 "F:\\nx3\\src\\nx3\\NVoice.hx"
						return ((global::haxe.lang.Function) (new global::haxe.lang.Closure(((object) (this) ), ((string) ("get_length") ), ((int) (261031087) ))) );
					}
					
					
					case 520590566:
					{
						#line 11 "F:\\nx3\\src\\nx3\\NVoice.hx"
						if (handleProperties) {
							#line 11 "F:\\nx3\\src\\nx3\\NVoice.hx"
							return this.get_length();
						}
						 else {
							#line 11 "F:\\nx3\\src\\nx3\\NVoice.hx"
							return this.length;
						}
						
					}
					
					
					case 328878574:
					{
						#line 11 "F:\\nx3\\src\\nx3\\NVoice.hx"
						return ((global::haxe.lang.Function) (new global::haxe.lang.Closure(((object) (this) ), ((string) ("iterator") ), ((int) (328878574) ))) );
					}
					
					
					case 543535969:
					{
						#line 11 "F:\\nx3\\src\\nx3\\NVoice.hx"
						return this.npart;
					}
					
					
					case 1292432058:
					{
						#line 11 "F:\\nx3\\src\\nx3\\NVoice.hx"
						return this.type;
					}
					
					
					case 453807379:
					{
						#line 11 "F:\\nx3\\src\\nx3\\NVoice.hx"
						return this.nnotes;
					}
					
					
					case 272654911:
					{
						#line 11 "F:\\nx3\\src\\nx3\\NVoice.hx"
						return this.direction;
					}
					
					
					default:
					{
						#line 11 "F:\\nx3\\src\\nx3\\NVoice.hx"
						return base.__hx_getField(field, hash, throwErrors, isCheck, handleProperties);
					}
					
				}
				
			}
			#line default
		}
		
		
		public override   double __hx_getField_f(string field, int hash, bool throwErrors, bool handleProperties){
			unchecked {
				#line 11 "F:\\nx3\\src\\nx3\\NVoice.hx"
				switch (hash){
					case 520590566:
					{
						#line 11 "F:\\nx3\\src\\nx3\\NVoice.hx"
						if (handleProperties) {
							#line 11 "F:\\nx3\\src\\nx3\\NVoice.hx"
							return ((double) (this.get_length()) );
						}
						 else {
							#line 11 "F:\\nx3\\src\\nx3\\NVoice.hx"
							return ((double) (this.length) );
						}
						
					}
					
					
					default:
					{
						#line 11 "F:\\nx3\\src\\nx3\\NVoice.hx"
						return base.__hx_getField_f(field, hash, throwErrors, handleProperties);
					}
					
				}
				
			}
			#line default
		}
		
		
		public override   object __hx_invokeField(string field, int hash, global::Array dynargs){
			unchecked {
				#line 11 "F:\\nx3\\src\\nx3\\NVoice.hx"
				switch (hash){
					case 1775117066:
					{
						#line 11 "F:\\nx3\\src\\nx3\\NVoice.hx"
						return this.getNNote(((int) (global::haxe.lang.Runtime.toInt(dynargs[0])) ));
					}
					
					
					case 261031087:
					{
						#line 11 "F:\\nx3\\src\\nx3\\NVoice.hx"
						return this.get_length();
					}
					
					
					case 328878574:
					{
						#line 11 "F:\\nx3\\src\\nx3\\NVoice.hx"
						return this.iterator();
					}
					
					
					default:
					{
						#line 11 "F:\\nx3\\src\\nx3\\NVoice.hx"
						return base.__hx_invokeField(field, hash, dynargs);
					}
					
				}
				
			}
			#line default
		}
		
		
		public override   void __hx_getFields(global::Array<object> baseArr){
			unchecked {
				#line 11 "F:\\nx3\\src\\nx3\\NVoice.hx"
				baseArr.push("length");
				#line 11 "F:\\nx3\\src\\nx3\\NVoice.hx"
				baseArr.push("npart");
				#line 11 "F:\\nx3\\src\\nx3\\NVoice.hx"
				baseArr.push("type");
				#line 11 "F:\\nx3\\src\\nx3\\NVoice.hx"
				baseArr.push("nnotes");
				#line 11 "F:\\nx3\\src\\nx3\\NVoice.hx"
				baseArr.push("direction");
				#line 11 "F:\\nx3\\src\\nx3\\NVoice.hx"
				{
					#line 11 "F:\\nx3\\src\\nx3\\NVoice.hx"
					base.__hx_getFields(baseArr);
				}
				
			}
			#line default
		}
		
		
	}
}

File: Nx3Java/java/src/nx3/NVoice.java
package nx3;
import haxe.root.*;

@SuppressWarnings(value={"rawtypes", "unchecked"})
public  class NVoice extends haxe.lang.HxObject
{
	public    NVoice(haxe.lang.EmptyObject empty)
	{
		{
		}
		
	}
	
	
	public    NVoice(haxe.root.Array<nx3.NNote> notes, nx3.EVoiceType type, nx3.EDirectionUAD direction)
	{
		nx3.NVoice.__hx_ctor_nx3_NVoice(this, notes, type, direction);
	}
	
	
	public static   void __hx_ctor_nx3_NVoice(nx3.NVoice __temp_me76, haxe.root.Array<nx3.NNote> notes, nx3.EVoiceType type, nx3.EDirectionUAD direction)
	{
		if (( ( notes == null ) || ( notes.length == 0 ) )) 
		{
			__temp_me76.nnotes = new haxe.root.Array<nx3.NNote>(new nx3.NNote[]{});
			__temp_me76.type = nx3.EVoiceType.Barpause(0);
		}
		 else 
		{
			__temp_me76.nnotes = notes;
			if (( type != null )) 
			{
				__temp_me76.type = type;
			}
			 else 
			{
				__temp_me76.type = nx3.EVoiceType.Normal;
			}
			
		}
		
		{
			int _g = 0;
			while (( _g < notes.length ))
			{
				nx3.NNote note = notes.__get(_g);
				 ++ _g;
				note.nvoice = __temp_me76;
			}
			
		}
		
		if (( direction != null )) 
		{
			__temp_me76.direction = direction;
		}
		 else 
		{
			__temp_me76.direction = nx3.EDirectionUAD.Auto;
		}
		
	}
	
	
	public static   java.lang.Object __hx_createEmpty()
	{
		return new nx3.NVoice(((haxe.lang.EmptyObject) (haxe.lang.EmptyObject.EMPTY) ));
	}
	
	
	public static   java.lang.Object __hx_create(haxe.root.Array arr)
	{
		return new nx3.NVoice(((haxe.root.Array<nx3.NNote>) (arr.__get(0)) ), ((nx3.EVoiceType) (arr.__get(1)) ), ((nx3.EDirectionUAD) (arr.__get(2)) ));
	}
	
	
	public  nx3.EDirectionUAD direction;
	
	public  haxe.root.Array<nx3.NNote> nnotes;
	
	public  nx3.EVoiceType type;
	
	public  nx3.NPart npart;
	
	public   java.lang.Object iterator()
	{
		return ((java.lang.Object) (new _Array.ArrayIterator<nx3.NNote>(((haxe.root.Array<nx3.NNote>) (this.nnotes) ))) );
	}
	
	
	public  int length;
	
	public   int get_length()
	{
		return this.nnotes.length;
	}
	
	
	public   nx3.NNote getNNote(int idx)
	{
		if (( ( idx < 0 ) || ( idx > this.nnotes.length ) )) 
		{
			return null;
		}
		 else 
		{
			return this.nnotes.__get(idx);
		}
		
	}
	
	
	@Override public   double __hx_setField_f(java.lang.String field, double value, boolean handleProperties)
	{
		{
			boolean __temp_executeDef598 = true;
			switch (field.hashCode())
			{
				case -1106363674:
				{
					if (field.equals("length")) 
					{
						__temp_executeDef598 = false;
						this.length = ((int) (value) );
						return value;
					}
					
					break;
				}
				
				
			}
			
			if (__temp_executeDef598) 
			{
				return super.__hx_setField_f(field, value, handleProperties);
			}
			 else 
			{
				throw null;
			}
			
		}
		
	}
	
	
	@Override public   java.lang.Object __hx_setField(java.lang.String field, java.lang.Object value, boolean handleProperties)
	{
		{
			boolean __temp_executeDef599 = true;
			switch (field.hashCode())
			{
				case -1106363674:
				{
					if (field.equals("length")) 
					{
						__temp_executeDef599 = false;
						this.length = ((int) (haxe.lang.Runtime.toInt(value)) );
						return value;
					}
					
					break;
				}
				
				
				case -962590849:
				{
					if (field.equals("direction")) 
					{
						__temp_executeDef599 = false;
						this.direction = ((nx3.EDirectionUAD) (value) );
						return value;
					}
					
					break;
				}
				
				
				case 105020769:
				{
					if (field.equals("npart")) 
					{
						__temp_executeDef599 = false;
						this.npart = ((nx3.NPart) (value) );
						return value;
					}
					
					break;
				}
				
				
				case -1040751853:
				{
					if (field.equals("nnotes")) 
					{
						__temp_executeDef599 = false;
						this.nnotes = ((haxe.root.Array<nx3.NNote>) (value) );
						return value;
					}
					
					break;
				}
				
				
				case 3575610:
				{
					if (field.equals("type")) 
					{
						__temp_executeDef599 = false;
						this.type = ((nx3.EVoiceType) (value) );
						return value;
					}
					
					break;
				}
				
				
			}
			
			if (__temp_executeDef599) 
			{
				return super.__hx_setField(field, value, handleProperties);
			}
			 else 
			{
				throw null;
			}
			
		}
		
	}
	
	
	@Override public   java.lang.Object __hx_getField(java.lang.String field, boolean throwErrors, boolean isCheck, boolean handleProperties)
	{
		{
			boolean __temp_executeDef600 = true;
			switch (field.hashCode())
			{
				case 1959846858:
				{
					if (field.equals("getNNote")) 
					{
						__temp_executeDef600 = false;
						return ((haxe.lang.Function) (new haxe.lang.Closure(((java.lang.Object) (this) ), haxe.lang.Runtime.toString("getNNote"))) );
					}
					
					break;
				}
				
				
				case -962590849:
				{
					if (field.equals("direction")) 
					{
						__temp_executeDef600 = false;
						return this.direction;
					}
					
					break;
				}
				
				
				case 974314479:
				{
					if (field.equals("get_length")) 
					{
						__temp_executeDef600 = false;
						return ((haxe.lang.Function) (new haxe.lang.Closure(((java.lang.Object) (this) ), haxe.lang.Runtime.toString("get_length"))) );
					}
					
					break;
				}
				
				
				case -1040751853:
				{
					if (field.equals("nnotes")) 
					{
						__temp_executeDef600 = false;
						return this.nnotes;
					}
					
					break;
				}
				
				
				case -1106363674:
				{
					if (field.equals("length")) 
					{
						__temp_executeDef600 = false;
						if (handleProperties) 
						{
							return this.get_length();
						}
						 else 
						{
							return this.length;
						}
						
					}
					
					break;
				}
				
				
				case 3575610:
				{
					if (field.equals("type")) 
					{
						__temp_executeDef600 = false;
						return this.type;
					}
					
					break;
				}
				
				
				case 1182533742:
				{
					if (field.equals("iterator")) 
					{
						__temp_executeDef600 = false;
						return ((haxe.lang.Function) (new haxe.lang.Closure(((java.lang.Object) (this) ), haxe.lang.Runtime.toString("iterator"))) );
					}
					
					break;
				}
				
				
				case 105020769:
				{
					if (field.equals("npart")) 
					{
						__temp_executeDef600 = false;
						return this.npart;
					}
					
					break;
				}
				
				
			}
			
			if (__temp_executeDef600) 
			{
				return super.__hx_getField(field, throwErrors, isCheck, handleProperties);
			}
			 else 
			{
				throw null;
			}
			
		}
		
	}
	
	
	@Override public   double __hx_getField_f(java.lang.String field, boolean throwErrors, boolean handleProperties)
	{
		{
			boolean __temp_executeDef601 = true;
			switch (field.hashCode())
			{
				case -1106363674:
				{
					if (field.equals("length")) 
					{
						__temp_executeDef601 = false;
						if (handleProperties) 
						{
							return ((double) (this.get_length()) );
						}
						 else 
						{
							return ((double) (this.length) );
						}
						
					}
					
					break;
				}
				
				
			}
			
			if (__temp_executeDef601) 
			{
				return super.__hx_getField_f(field, throwErrors, handleProperties);
			}
			 else 
			{
				throw null;
			}
			
		}
		
	}
	
	
	@Override public   java.lang.Object __hx_invokeField(java.lang.String field, haxe.root.Array dynargs)
	{
		{
			boolean __temp_executeDef602 = true;
			switch (field.hashCode())
			{
				case 1959846858:
				{
					if (field.equals("getNNote")) 
					{
						__temp_executeDef602 = false;
						return this.getNNote(((int) (haxe.lang.Runtime.toInt(dynargs.__get(0))) ));
					}
					
					break;
				}
				
				
				case 1182533742:
				{
					if (field.equals("iterator")) 
					{
						__temp_executeDef602 = false;
						return this.iterator();
					}
					
					break;
				}
				
				
				case 974314479:
				{
					if (field.equals("get_length")) 
					{
						__temp_executeDef602 = false;
						return this.get_length();
					}
					
					break;
				}
				
				
			}
			
			if (__temp_executeDef602) 
			{
				return super.__hx_invokeField(field, dynargs);
			}
			 else 
			{
				throw null;
			}
			
		}
		
	}
	
	
	@Override public   void __hx_getFields(haxe.root.Array<java.lang.String> baseArr)
	{
		baseArr.push("length");
		baseArr.push("npart");
		baseArr.push("type");
		baseArr.push("nnotes");
		baseArr.push("direction");
		{
			super.__hx_getFields(baseArr);
		}
		
	}
	
	
}


